src/features/island/buildings/components/building/WithCraftingMachine.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext, useState } from "react";
+import { useInterpret, useSelector } from "@xstate/react";
+import isEmpty from "lodash.isempty";
+import { Context } from "features/game/GameProvider";
+import {
+  CraftingContext,
+  craftingMachine,
+  MachineInterpreter,
+  MachineState,
+} from "../../lib/craftingMachine";
+import { CraftingTimerModal } from "../ui/CraftingTimerModal";
+import { BuildingProps } from "./Building";
+import { ConsumableName } from "features/game/types/consumables";
+
+const isIdle = (state: MachineState) => state.matches("idle");
+const isCrafting = (state: MachineState) => state.matches("crafting");
+const isReady = (state: MachineState) => state.matches("ready");
+const itemName = (state: MachineState) => state.context.name;
+
+export interface CraftingMachineChildProps extends BuildingProps {
+  idle: boolean;
```

Comment:
Instead of 3 separate states could we simplify this into a single state?

```
status: "idle" | "crafting" | "ready"
```

This will also help prevent some invalid states from being created.

E.g.
```
idle: true;
crafting: true;
ready: false;
```
This would satisfy the types but shouldn't happen in the game.